File: practica2_844759_845647/c++/write.cc
//------------------------------------------------------------------------------
// File:   write.cc
// Authors: Jorge Leris - 845647 / Angel Villanueva - 844759
// Date:   febrero 2023
// Coms:   Fichero "read" de la práctica 2
//------------------------------------------------------------------------------
#include "write.h"

void Write::ejecutar(stack<int> &pila, int &contador) const
{
    int dato = pila.top();
    pila.pop();
    cout << dato << endl;
    contador++;
}

void Write::listar() const
{
    cout << "write" << endl;
}

File: practica2_844759_845647/java/Write.java
//------------------------------------------------------------------------------
// File:   write.java
// Authors: Jorge Leris - 845647 / Angel Villanueva - 844759
// Date:   febrero 2023
// Coms:   Práctica 2 de TecPro
//------------------------------------------------------------------------------

import java.util.Stack;

public class Write extends Instruccion
{
    public void ejecutar(Stack<Integer> Pila, int contador){
        int dato = pila.peek();
        pila.pop();
        System.out.println(dato);
        contador = contador + 1;
    }

    public void listar(){
        System.out.println("write");
    }
}